pkg/fn/eval.go

```diff
@@ -17,13 +17,19 @@ package fn
 import (
 	"context"
 
+	"github.com/GoogleContainerTools/kpt/internal/types"
+	fnresult "github.com/GoogleContainerTools/kpt/pkg/api/fnresult/v1"
 	v1 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1"
 	"sigs.k8s.io/kustomize/kyaml/filesys"
+	"sigs.k8s.io/kustomize/kyaml/kio"
 )
 
 type EvalOptions struct {
+	// FnResultList stores the result of the function evaluation
+	FnResultList fnresult.ResultList
 }
 
 type Evaluator interface {
 	Eval(ctx context.Context, pkg filesys.FileSystem, fn v1.Function, opts EvalOptions) error
```

Comment:
@martinmaly Do we need this ? Can't we just leverage NewRunner function and kio.Filter interface ? Refer to fnruntime package.

Reply:
it may work. What does pkgPath do? If we use kio.Filter there should be no need for paths.

Reply:
pkgPath can be used for printing status messages. Removing it for now, you can reintroduce it based on the implementation.